contracts/AirDrop.sol

```diff
@@ -0,0 +1,242 @@
+        require(_owner == msgSigner, "invalid signature");
+        require(
+            block.timestamp < deadline_,
+            "Error : signed transaction expired"
+        );
+
+        balanceOfTokens[recepients_] += amount_;
+        token.approve(recepients_, amount_);
+        emit DropTokens(recepients_, amount_, deadline_);
+    }
+
+    /**
+     * @dev make ethers available for members to withdraw
+     *
+     * @param recepients_ - address of member
+     * @param amount_ - how many ethers will be available
+     * @param deadline_ - how long signature will be valid
+     * @param v_ - can be only 27 or 28
+     * @param r_ - contain info to recover signer
+     * @param s_ - contain info to recover signer
+     *
+     * NOTE : more information 'EIP712'
+     */
+    function dropEther(
+        address recepients_,
+        uint256 amount_,
+        uint256 deadline_,
+        uint8 v_,
+        bytes32 r_,
+        bytes32 s_
+    ) external override onlyOwner {
+        require(
+            amount_ != 0 && recepients_ != address(0) && deadline_ != 0,
+            "Error :'recepients_' or 'amount_' or 'deadline_' equal to 0"
+        );
+        bytes32 structHash = keccak256(
+            abi.encode(DROP_HASH, amount_, deadline_, recepients_)
+        );
+        bytes32 digest = EIP712._hashTypedDataV4(structHash);
+
+        address msgSigner = ECDSA.recover(digest, v_, r_, s_);
+        require(_owner == msgSigner, "invalid signature");
+        require(
+            block.timestamp < deadline_,
+            "Error : signed transaction expired"
+        );
+
+        balanceOfEther[recepients_] += amount_;
+        emit DropEther(recepients_, amount_, deadline_);
+    }
+
+    /**
+     * @dev Change actually ERC20
+     *
+     * @param token_ - address of new ERC20 which connected
+     * to 'this' contract
+     */
+    function updateTokenAddress(address token_) external override onlyOwner {
+        require(
+            token_.code.length > 0,
+            "Error : Incorrect address , only contract address"
+        );
+        token = IERC20(token_);
+        emit NewContractAddress(token_);
+    }
+
+    /**
+     * @dev This function return tokens to 'owner'
+     *
+     * NOTE: withdraw all available tokens
+     */
+    function withdrawTokens() external override onlyOwner {
+        uint256 balance = token.balanceOf(address(this));
+        require(balance > 0, "Error : No availabale tokens to withdraw");
+        emit WithdrawTokens(balance, msg.sender);
+        token.safeTransfer(msg.sender, balance);
+    }
+
+    /**
+     * @dev This function return ether to 'owner'
+     *
+     * NOTE: withdraw all available ether
+     */
+    function withdrawEther() external override onlyOwner {
+        require(
+            address(this).balance > 0,
+            "Error : No availabale ether to withdraw"
+        );
+        emit WithdrawEther(address(this).balance, msg.sender);
```

Comment:
event should be in end of functiona, incorrect order

Reply:
also move msg.sender to first place,
use next pattern:
Event(sender, recipient, anotherData...)